SocketCAN/rxServers.py

```diff
@@ -0,0 +1,114 @@
+import socket
+import uuid
+from multiprocessing import Process
+import struct
+import sys
+
+SERVER_IP = "127.0.0.1"
+try:
+	SERVER_IP = socket.gethostbyname(socket.gethostname())
+except Exception as e:
+	print("Could not get this machine's IP address, defaulting to 127.0.0.1")
+
+# The basic CAN frame structure and the sockaddr structure are defined
+#   in include/linux/can.h:
+#     struct can_frame {
+#             canid_t can_id;  /* 32 bit CAN_ID + EFF/RTR/ERR flags */
+#             __u8    can_dlc; /* frame payload length in byte (0 .. 8) */
+#             __u8    __pad;   /* padding */
+#             __u8    __res0;  /* reserved / padding */
+#             __u8    __res1;  /* reserved / padding */
+#             __u8    data[8] __attribute__((aligned(8)));
+#     };   
+
+# To match this data structure, the following struct format can be used:
+can_frame_format = "<LL8s"
+# Unsigned Long Integer (little endian), unsigned char (byte), three pad bytes, eight chars (bytes)
+# Note: this is 16 bytes
+
+
+# These are defined in can.h of the Linux kernel sources
+# e.g. https://github.com/torvalds/linux/blob/master/include/uapi/linux/can.h
+"""print("socket.CAN_EFF_MASK = 0x{:08X}".format(socket.CAN_EFF_MASK))
+print("socket.CAN_EFF_FLAG = 0x{:08X}".format(socket.CAN_EFF_FLAG))
+print("socket.CAN_RTR_FLAG = 0x{:08X}".format(socket.CAN_RTR_FLAG))
+print("socket.CAN_ERR_FLAG = 0x{:08X}".format(socket.CAN_ERR_FLAG))
+print("socket.CAN_ERR_FLAG = 0x{:08X}".format(socket.CAN_SFF_MASK))"""
+
+CAN_EFF_MASK = 0x1FFFFFFF #assigned to remove socketCAN dependencies, values obtained by above print statements using socketCAN
+CAN_EFF_FLAG = -0x80000000
```

Comment:
Why is this negative? We should be using unsigned integers.

Reply:
I'll fix this. It was a straight copy/paste from what was printed when using the CAN_EFF_FLAG from socketCAN. Thanks for the catch.